src/doc/trpl/the-stack-and-the-heap.md

```diff
@@ -1,3 +1,526 @@
+| 0x01       | i    | 0x40000000 |
+| 0x00       | h    | 3          |
+
+Regardless, you may see diagrams in other examples that use hexadecimal. We’ll
+just stick to decimal, since we’re already halfway through the example.
+
+At the end of `bar()`, it calls `baz()`:
+
+| Address      | Name | Value        |
++--------------+------+--------------+
+|  2 ^ 30      |      | 20           |
+| (2 ^ 30) - 1 |      | 5            |
+| ...          | ...  | ...          |
+| 12           | g    | 100          |
+| 11           | f    | 4            |
+| 10           | e    | 4            |
+| 9            | d    | (2 ^ 30) - 1 |
+| 8            | c    | 5            |
+| 7            | b    | 4            |
+| 6            | a    | 0            |
+| 5            | z    | 4            |
+| 4            | y    | 10           |
+| 3            | x    | 0            |
+| 2            | j    | 0            |
+| 1            | i    | 2 ^ 30       |
+| 0            | h    | 3            |
+
+With this, we’re at our deepest point! Whew! Congrats for following along this
+far.
+
+After `baz()` is over, we get rid of `f` and `g`:
+
+| Address      | Name | Value        |
++--------------+------+--------------+
+|  2 ^ 30      |      | 20           |
+| (2 ^ 30) - 1 |      | 5            |
+| ...          | ...  | ...          |
+| 10           | e    | 4            |
+| 9            | d    | (2 ^ 30) - 1 |
+| 8            | c    | 5            |
+| 7            | b    | 4            |
+| 6            | a    | 0            |
+| 5            | z    | 4            |
+| 4            | y    | 10           |
+| 3            | x    | 0            |
+| 2            | j    | 0            |
+| 1            | i    | 2 ^ 30       |
+| 0            | h    | 3            |
+
+Next, we return from `bar()`. `d` in this case is a `Box<T>`, so it also frees
+what it points to: `(2 ^ 30) - 1`.
+
+| Address | Name | Value  |
++---------+------+--------+
+|  2 ^ 30 |      | 20     |
+| ...     | ...  | ...    |
+| 5       | z    | 4      |
+| 4       | y    | 10     |
+| 3       | x    | 0      |
+| 2       | j    | 0      |
+| 1       | i    | 2 ^ 30 |
+| 0       | h    | 3      |
+
+And after that, `foo()` returns:
+
+| Address | Name | Value  |
++--------------+------+---+
+|  2 ^ 30 |      | 20     |
+| ...     | ...  | ...    |
+| 2       | j    | 0      |
+| 1       | i    | 2 ^ 30 |
+| 0       | h    | 3      |
+
+And then, finally, `main()`, which cleans the rest up. When `i` is `Drop`ped,
+it will clean up the last of the heap too.
+
+# What do other languages do?
+
+Most languages with a garbage collector heap-allocate by default. This means
+that every value is boxed. There are a number of reasons why this is done, but
+they’re out of scope for this tutorial. There are some possible optimizations
+that don’t make it true 100% of the time, too. Rather than relying on the stack
+and `Drop` to clean up memory, the garbage collector deals with the heap
+instead.
```

Comment:
Yeah. i'm gonna adapt this text into a 'why is the stack neccesary' section. Thanks :)